scripts/src/main/resources/scripts/command/docker

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+wsl_linux_version=$(wsl -l)
```

Comment:
even though we do not support other OS than windows yet, could you please alread add OS checks as suggested in last review?
```
if doIsWindows
then
   # ... add your code here
else
  doFail "Sorry, docker-installation support is not yet implemented for your OS. Please install manually or help devonfw-ide to support it for your OS by contributing a pull-request."
fi
```

Further, I suggested to design these commandlets following the philosophy of other commandlets.
So what you are doing here should only happen for `devon docker setup`.
For any other command (excluding `help`) you should delegate to an existing docker installation and pass the arguments to the docker command.
Finally, we always do a setup in a lazy way. So currently I can run
```
devon mvn clean install
```
And if `mvn` is not yet installed, it will be installed first.
At the end (both if mvn was already present or just has been installed) `mvn clean install` is called.

You should be able to check if `docker` command is present in an OS independent way.
Then it can already be used on all OS but only the automatic installation does not work on OS other than windows.

Please have a look at existing commandlets like `mvn`, `npm`, `java`, etc. and follow our existing philosophy.

Reply:
I think as docker or kubernetes are global services, it does not really make sense to proxy the commands by devon command.
I even did not understood why we are providing this anyhow, I am not using it at all and it makes me feel even more to be working with a framework if I am told to use `devon mvn *` instead of `mvn *`.

Reply:
> even though we do not support other OS than windows yet, could you please alread add OS checks as suggested in last review?
> 
> ```
> if doIsWindows
> then
>    # ... add your code here
> else
>   doFail "Sorry, docker-installation support is not yet implemented for your OS. Please install manually or help devonfw-ide to support it for your OS by contributing a pull-request."
> fi
> ```
> 
> Further, I suggested to design these commandlets following the philosophy of other commandlets.
> So what you are doing here should only happen for `devon docker setup`.
> For any other command (excluding `help`) you should delegate to an existing docker installation and pass the arguments to the docker command.
> Finally, we always do a setup in a lazy way. So currently I can run
> 
> ```
> devon mvn clean install
> ```
> 
> And if `mvn` is not yet installed, it will be installed first.
> At the end (both if mvn was already present or just has been installed) `mvn clean install` is called.
> 
> You should be able to check if `docker` command is present in an OS independent way.
> Then it can already be used on all OS but only the automatic installation does not work on OS other than windows.
> 
> Please have a look at existing commandlets like `mvn`, `npm`, `java`, etc. and follow our existing philosophy.

Have incorporated the OS checks now.

Reply:
Nobody is forced to do `devon mvn *` instead of `mvn *`. However, this allows scripts to be portable and work seamlessly. Otherwise, you would have to write something like:
```
if ! command -v mvn &> /dev/null
then
  devon mvn setup
fi
mvn my-goal
```

This might still not be enough if `mvn` is installed but the devon environment variables have not been set accordingly.
Instead a script can also do
```
devon mvn my-goal
```
and it will always work and even install maven on the fly if not yet available.

Aint the entire goal of our docker and kubernetes support that projects could also provide scripts to provide e.g. a database running via container as a most simple script in a portal and automated way?

If you want we can have a fundamental discussion about the philopsophy of devonfw-ide and if you think that I designed all odd and wrong. However, we IMHO here want to discuss about adding these two commandlets.
I am repeating myself, but I do not accept that all existing commandlets follow a common philosophy for IMHO good reasons and that these two new commandelts should ignore them what would also is required for specific features to work. E.g. you currently can not add `docker` or `kubernetes` to `DEVON_IDE_TOOLS` variable because of that.